### ISSUE TITLE
no-useless-escape parses regexes incorrectly

### ISSUE THREAD

**Issue by fabiospampinato** [eslint/eslint#12408]
**Tell us about your environment**

* **ESLint Version:** v6.5.1
* **Node Version:** v10.15.0
* **npm Version:** v6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**

@typescript-eslint/parser

**Please show your full configuration:**

This should reproduce it:

```json
{
  "env": {
    "browser": true,
    "es6": true,
    "node": true
  },
  "globals": {
    "__static": "readonly",
    "$": "readonly",
    "Svelto": "readonly"
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": 2020,
    "project": "tsconfig.json",
    "sourceType": "module",
    "useJSXTextNode": true
  },
  "settings": {
    "react": {
      "pragma": "React",
      "version": "detect"
    }
  },
  "plugins": [
    "react",
    "react-hooks",
    "@typescript-eslint",
    "unicorn"
  ],
  "rules": {
    "no-useless-escape": "error"
  }
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/\//g
```

```bash
npx eslint src --ext '.ts, .tsx'
```

**What did you expect to happen?**

No errors.

**What actually happened? Please include the actual, raw output from ESLint.**

The linter rule is saying that I shouldn't be escaping the slash in the regex.

**Are you willing to submit a pull request to fix this bug?**

It depends I guess 🤷‍♂️

I'm not sure this is actually an eslint bug, but most probably a bug in whatever parser you guys are using, or typescript's parser that I'm using.

If I put `///g` into [Esprima](https://esprima.org/demo/parse.html?code=%2F%E2%88%95%2Fg%0A) it actually thinks that that's a valid regex, but when I run the code in Node's REPL I get the following:

```js
///g.test // => undefined
/\//g.test // => [Function: test]
console.log ( ///g.test ) // ..., it is waiting for more code, weird
```

**Comment by mdjermanovic** [comment]
Hi @fabiospampinato, thanks for the issue!

I can't reproduce the error with the default parser (espree) in the demo - [Online Demo Link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVzZWxlc3MtZXNjYXBlOiBlcnJvciAqL1xuXG52YXIgZm9vID0gL1xcLy9nIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==) 

I noticed two interesting things in your examples:

* The Esprima link - the middle character looks like slash, but it isn't: `/∕/g`. Maybe you got the useless escape warning on that character?
* In some of your examples `//` is start of a line comment, not regex.

**Comment by fabiospampinato** [comment]
@mdjermanovic You're absolutely right, that's not a slash, '/' !== '∕' 🤦‍♂️ Thanks!
